src/schema/organization/contributors/resolvers.js

```diff
@@ -1,61 +1,131 @@
+    const reviewsNodes = user.contributionsCollection.pullRequestReviewContributions.nodes;
+    const contributions = [issuesNodes, pullRequestsNodes, reviewsNodes];
+
+    const dates = [];
+    var firstContrib;
+    contributions.forEach(contribution => {
+      if (contribution && contribution.length > 0) {
+        firstContrib = contribution[0].occurredAt;
+        dates.push(new Date(firstContrib)); 
+      }
+    });
+
+    var result = undefined;
+    if (dates.length > 0)
+      result = getOldestDate(dates);
+    return result;
+  }
+}
+
+const firstContributionDateResolver = async(parent, args, { token }) => {
+  const { login } = args;
+  const body = githubQueries.firstContributionDate(login, parent.id);
+  const data = await fetchData(body, token);
+  
+  var result = processFirstContributionDate(data.data.user);
+  return result;
+}
+
+const validateLogin = async (login, orgID, token) => {
+  const body = githubQueries.isContributor(login, orgID);
+  const data = await fetchData(body, token);
+  const { user } = data.data;
+  const isValid = (user && user.contributionsCollection.hasAnyContributions);
+  return isValid;
+}
+
 export default {
   Query: {
     contributors: async (parent, args, { token }) => {
       const { first, last, after, before } = args;
-      const { id: orgID } = parent;
-      
-      // TODO: Fazer outra consulta se totalRepos > 100
+
       const repoArgs = { first: parent.totalRepos }; 
       const userArgs = { first };
 
-      // TODO: Otimizar verificando o número de contribuidores obtidos por iteração em repositorio
-      // caso a quantidade já passe de `first`, paramos de iterar.
       const body = githubQueries.contributors(parent, repoArgs, userArgs);
       const data = await fetchData(body, token);
-
-      let contributorsArray = process(data.data.organization.repositories.nodes);
+      
+      var contributorsArray = processContributors(data.data.organization.repositories.nodes);
       contributorsArray.length = first;
       return contributorsArray;
     },
+    contributor: async (parent, args, { token }, info) => {
+      const { login } = args;
+      var loginIsValid = await validateLogin(login, parent.id, token);
+      var contributor;
+
+      if (loginIsValid) {
+        const body = githubQueries.contributor(login);
+        const data = await fetchData(body, token);
+        contributor = data.data.user;
+  
+        const { fieldNodes } = info;
+        const rootFields = (fieldNodes.filter(node => node.name.value == 'contributor')).pop();
+        const fields = rootFields.selectionSet.selections.map(node => node.name.value);
+  
+        if (fields.includes('firstContributionDate')) {
+          contributor.firstContributionDate = firstContributionDateResolver(parent, args, { token });
+        }
+      } else {
+        throw new Error(`User '${login}' may not be a GitHub User or a Contributor.`
+        );
```

Comment:
The identation is weird

Reply:
I guess was the width of the review page. Seems very normal to me, but you can look at https://github.com/panelinhadees/server/blob/refactor/95-contributors-query/src/schema/organization/contributors/resolvers.js